### PR TITLE
Feature/add git hooks

### PR DESCRIPTION
**Pull request by mattch1**
Add precommit script to check feature flags and versions.

### CODE REVIEW

File: .githooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env sh
+exec < /dev/tty
+    while read -p "Have all the required feature flags been added to these changes? (y/n) " yn; do
+        case $yn in
+            [Yy] ) break;;
+            [Nn] ) echo "Please add any necessary feature flags!"; exit 1;;
```
Comment (mwestacott) [inline .githooks/pre-commit]: Is the double ;; at end of line needed?

File: .githooks/pre-commit [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env sh
+exec < /dev/tty
```
Comment (mwestacott) [inline .githooks/pre-commit]: Guessing that we'll add the 'npm run lint' command later?

File: .githooks/pre-commit [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        esac
+    done
+    while read -p "Is the version file up to date? (y/n) " yn; do
+        case $yn in
+            [Yy] ) break;;
+            [Nn] ) echo "If required please update the version file!"; exit 1;;
+            * ) echo "Please answer y (yes) or n (no):" && continue;
+        esac
+    done
+exec
```
Comment (mwestacott) [inline .githooks/pre-commit]: Also wondering why extensions has '<&-' at the end of this line. Would it be needed here?

File: .githooks/pre-commit (hunk 2 above)
Comment (mattch1) [reply .githooks/pre-commit]: Yes that'll be when we add the node js code with the lint library.

File: .githooks/pre-commit (hunk 1 above)
Comment (mattch1) [reply .githooks/pre-commit]: In this code doule ;; are required.

**Pull request merged**